### PR TITLE
feat: add effort directive

### PR DESCRIPTION
**Pull request by moorejs**
- **Quick Checklist**

* [x] I have read [the contributing guidelines](../CONTRIBUTING.md)
* [x] I have written new tests, as applicable (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] I have added a changeset, if applicable

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

New directive to control the encoding effort.

- **What is the new behavior (if this is a feature change)?**

Users can specify efforts values that differ from the default effort for a given format.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this
  PR?)

No

- **Other information**:

`"min"` and `"max"` keywords create a dependency on the format value in the metadata of the image. This was a compromise since specifying an exact number doesn't really make sense if you're generating multiple formats:
e.g. `?format=webp;avif&effort=7` would mean webp effort is default since it is out of range.

### CODE REVIEW

File: docs/directives.md [hunk 1]
```diff
@@ -85,6 +86,32 @@ import Image from 'example.jpg?blur=100'
...
+
+• **Keyword**: `effort`<br> • **Type**: _integer_ | _"max"_ | _"min"_ <br>
+
+Adjust the effort to spend encoding the image.
+The effect of effort varies per format, but a lower value leads to faster encoding.
+
+The supported ranges by format:
+- `png`: 1 to 10 (default 7)
+- `webp`: 0 to 6 (default 4)
+- `avif`/`heif`/`heic`: 0 to 9 (default 4)
```
Comment (benmccann) [inline docs/directives.md:99]: I don't think sharp supports `heic`: https://github.com/lovell/sharp/blob/eefaa998725cf345227d94b40615e090495c6d09/lib/index.d.ts#L1614
```suggestion
- `avif`/`heif`/ 0 to 9 (default 4)
```

File: packages/core/src/transforms/effort.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export interface EffortOptions {
+  effort: string
+}
+
+const FORMAT_TO_EFFORT_RANGE: Record<string, [number, number]> = {
+  png: [1, 10],
+  webp: [0, 6],
+  avif: [0, 9],
+  heif: [0, 9],
+  heic: [0, 9],
```
Comment (benmccann) [inline packages/core/src/transforms/effort.ts:13]:
```suggestion
  jxl: [3, 9],
```

**Comment by moorejs** [comment]
Thanks for the review. Generating docs appears to function now but the links are relative to my fork so I'll defer that.

**Review by benmccann — APPROVED** [review]

**Comment by benmccann** [comment]
Thank you! Sorry for the delay in merging this. I'll do a release after I get through a bit more of the PR queue

**Pull request merged**